sql/connect/common/src/main/protobuf/spark/connect/base.proto

```diff
@@ -1095,6 +1095,32 @@ message FetchErrorDetailsResponse {
     // Portable error identifier across SQL engines
     // If null, error class or SQLSTATE is not set.
     optional string sql_state = 4;
+
+    // Additional information if the error was caused by a breaking change.
+    optional BreakingChangeInfo breaking_change_info = 5;
+  }
+
+  // BreakingChangeInfo defines the schema for breaking change information.
+  message BreakingChangeInfo {
+    // A message explaining how the user can migrate their job to work
+    // with the breaking change.
+    repeated string migration_message = 1;
```

Comment:
shall we concatenate the string in the server side? For client's point of view, it's a bit weird to see the message as a list of string.

Reply:
It's a string list in JSON because we don't want to have super long lines in the JSON file. It's not really about semantic and I think we should not inherit this trick in the protobuf message.